common.js

```diff
@@ -74,6 +75,27 @@ function setopts (self, pattern, options) {
   self.statCache = options.statCache || Object.create(null)
   self.symlinks = options.symlinks || Object.create(null)
 
+  self.ignore = options.ignore || []
+
+  if(!Array.isArray(self.ignore)) {
+    self.ignore = [self.ignore]
+  }
+  if(self.ignore.length) {
+    self.ignore = self.ignore.map(function(pattern) {
```

Comment:
Oh, I see, this is already in common.  In that case, it can just be a module-private named function, if it's not needed externally.